### PR TITLE
impove(slb_listener) slb listener support same port.

### PR DESCRIPTION
**Pull request by xujunnnn**
=== RUN   TestAccAlicloudSlbListener_http_basic
--- PASS: TestAccAlicloudSlbListener_http_basic (74.76s)
PASS
ok      github.com/terraform-providers/terraform-provider-alicloud/alicloud     74.819s

=== RUN   TestAccCheckSlbListenerForward
--- PASS: TestAccCheckSlbListenerForward (18.21s)
PASS
ok      github.com/terraform-providers/terraform-provider-alicloud/alicloud     18.278s

=== RUN   TestAccAlicloudSlbListener_https_update
--- PASS: TestAccAlicloudSlbListener_https_update (69.56s)
PASS
ok      github.com/terraform-providers/terraform-provider-alicloud/alicloud     69.627s

=== RUN   TestAccAlicloudSlbListener_tcp_basic
--- PASS: TestAccAlicloudSlbListener_tcp_basic (49.62s)
PASS
ok      github.com/terraform-providers/terraform-provider-alicloud/alicloud     49.688s

=== RUN   TestAccAlicloudSlbListener_tcp_server_group
--- PASS: TestAccAlicloudSlbListener_tcp_server_group (136.34s)
PASS
ok      github.com/terraform-providers/terraform-provider-alicloud/alicloud     136.411s

=== RUN   TestAccAlicloudSlbListener_udp_basic
--- PASS: TestAccAlicloudSlbListener_udp_basic (41.66s)
PASS
ok      github.com/terraform-providers/terraform-provider-alicloud/alicloud     41.726s

=== RUN   TestAccAlicloudSlbListener_same_port
--- PASS: TestAccAlicloudSlbListener_same_port (16.46s)
PASS
ok      github.com/terraform-providers/terraform-provider-alicloud/alicloud     16.522s

### CODE REVIEW

File: alicloud/resource_alicloud_slb_listener.go [hunk 1]
```diff
@@ -376,7 +376,7 @@ func resourceAliyunSlbListenerCreate(d *schema.ResourceData, meta interface{}) e
 		return WrapErrorf(err, DefaultErrorMsg, "alicloud_slb_listener", request.GetActionName(), AlibabaCloudSdkGoERROR)
 	}
 	addDebug(request.GetActionName(), raw, request, request.QueryParams)
-	d.SetId(lb_id + ":" + strconv.Itoa(frontend))
+	d.SetId(lb_id + ":" + protocol + ":" + strconv.Itoa(frontend))
```
Comment (YY-tomorrow) [inline alicloud/resource_alicloud_slb_listener.go]: Please keep compatible

File: alicloud/resource_alicloud_slb_listener.go [hunk 2]
```diff
@@ -406,7 +407,7 @@ func resourceAliyunSlbListenerRead(d *schema.ResourceData, meta interface{}) err
 	client := meta.(*connectivity.AliyunClient)
 	slbService := SlbService{client}
 
-	lb_id, protocol, _, err := parseListenerId(d, meta)
+	lb_id, protocol, port, err := parseListenerId(d, meta)
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_slb_listener.go]: Please keep compatibility.

File: alicloud/common.go [hunk 3]
```diff
@@ -698,6 +698,14 @@ func ParseResourceId(id string, length int) (parts []string, err error) {
 	return parts, err
 }
 
+func ParseSlbListenerId(id string) (parts []string, err error) {
+	parts = strings.Split(id, ":")
+	if len(parts) != 2 && len(parts) != 3 {
+		err = WrapError(fmt.Errorf("Invalid alicloud_slb_listener Id %s.", id))
```
Comment (xiaozhu36) [inline alicloud/common.go]: Please add more message to guide users to set the correct resource id, like line 696.

File: alicloud/resource_alicloud_slb_listener.go [hunk 4]
```diff
@@ -417,7 +418,7 @@ func resourceAliyunSlbListenerRead(d *schema.ResourceData, meta interface{}) err
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_slb_listener.go]: Please set the new id after getting new parts.
In addition, please set the new id in the `Update` method.

File: alicloud/service_alicloud_slb.go [hunk 5]
```diff
@@ -152,7 +152,7 @@ func (s *SlbService) DescribeSlbBackendServer(id string) (*slb.DescribeLoadBalan
 }
 
 func (s *SlbService) DescribeSlbListener(id string, protocol Protocol) (listener map[string]interface{}, err error) {
```
Comment (xiaozhu36) [inline alicloud/service_alicloud_slb.go]: If the id has contained protocol, the method `DescribeSlbListener ` should only have one parameter `id`.

File: alicloud/common.go [hunk 6]
```diff
@@ -698,6 +698,14 @@ func ParseResourceId(id string, length int) (parts []string, err error) {
 	return parts, err
 }
 
+func ParseSlbListenerId(id string) (parts []string, err error) {
+	parts = strings.Split(id, ":")
+	if len(parts) != 2 && len(parts) != 3 {
+		err = WrapError(fmt.Errorf("Invalid alicloud_slb_listener Id %s. Expected Id like this: lb-xxxx:http:80.", id))
```
Comment (xiaozhu36) [inline alicloud/common.go]: `Expected Id like this: lb-xxxx:http:80` should be `Expected Id format is <slb id>:<protocol>:< frontend>`

File: alicloud/service_alicloud_common_test.go [hunk 7]
```diff
@@ -871,3 +871,101 @@ resource "alicloud_security_group" "default" {
...
+  available_disk_category = "cloud_efficiency"
+  available_resource_creation= "VSwitch"
+}
+
+data "alicloud_instance_types" "default" {
+  availability_zone = "${data.alicloud_zones.default.zones.0.id}"
+}
+
+data "alicloud_images" "default" {
+  name_regex = "^ubuntu_14.*_64"
```
Comment (xiaozhu36) [inline alicloud/service_alicloud_common_test.go]: Please use `"^ubuntu_18.*_64"`

**Pull request merged**